TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/CubePipeline.java

```diff
@@ -0,0 +1,164 @@
+package org.firstinspires.ftc.teamcode.hardware;
+
+import static java.util.Collections.emptyList;
+
+import android.graphics.Canvas;
+import android.graphics.Paint;
+
+import androidx.annotation.Nullable;
+
+import org.firstinspires.ftc.robotcore.internal.camera.calibration.CameraCalibration;
+import org.firstinspires.ftc.teamcode.utility.Api;
+import org.firstinspires.ftc.teamcode.utility.CubeSide;
+import org.firstinspires.ftc.teamcode.utility.FieldSide;
+import org.firstinspires.ftc.teamcode.utility.Pair;
+import org.firstinspires.ftc.vision.VisionProcessor;
+import org.opencv.core.Core;
+import org.opencv.core.CvType;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfPoint;
+import org.opencv.core.Point;
+import org.opencv.core.Scalar;
+import org.opencv.core.Size;
+import org.opencv.imgproc.Imgproc;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public class CubePipeline implements VisionProcessor {
```

Comment:
I think this is some pretty complex code, while in-line commenting would be nice, while we're at this stage of development let's not strain it, instead, I would like to see a comment header block describing what this class's purpose is, and then a different header block describing what each function does. A non-programmer should be able to read the header blocks of a function, and understand the general concept of what it is doing.

Reply:
done